pkgs/development/python-modules/amazon-ion/default.nix

```diff
@@ -23,15 +26,46 @@ buildPythonPackage rec {
     tag = "v${version}";
     # Test vectors require git submodule
     fetchSubmodules = true;
-    hash = "sha256-ZnslVmXE2YvTAkpfw2lbpB+uF85n/CvA22htO/Y7yWk=";
+    leaveDotGit = true; # During ion-c submodule build git history/hash used to infer version
+    hash = "sha256-wM4EybnRcIO4iblhxC3lPZuhG4CCUi04hSN4PIZyCnY=";
   };
 
+  patches = [
+    # backport changes that adapt code to more strict compilers
+    (fetchpatch2 {
+      name = "46aebb0-Address-incompatible-pointer-errors.patch";
+      url = "https://github.com/amazon-ion/ion-c/commit/46aebb0b650a4cf61425ef1a8e78e2443023e853.patch?full_index=1";
+      hash = "sha256-5qzSbZV9Oe5soJzkyCtVtWejedLEjAz7yuVotATPmbs=";
+      stripLen = 1;
+      extraPrefix = "ion-c/";
+    })
+  ];
+
   postPatch = ''
     substituteInPlace setup.py \
       --replace "'pytest-runner'," ""
+
+    # Ion C building is in _download_ion_c() which will try to remove sources and re-download
+    substituteInPlace install.py \
+      --replace-fail "isdir(_CURRENT_ION_C_DIR)" "False"
+    substituteInPlace install.py \
+      --replace-fail "check_call(['git'" "check_call(['true'"
   '';
 
-  build-system = [ setuptools ];
+  build-system = [
+    setuptools
+    cmake
```

Comment:
After adding `git` to `nativeBuildInputs` and reading [`buildPythonPackage` manual](https://nixos.org/manual/nixpkgs/stable/#buildpythonpackage-parameters) I'm no longer sure `cmake` belongs to `build-system`. But I see both examples. E.g. https://github.com/NixOS/nixpkgs/blob/a40440f13eb80881ab26734a36bb3005300e1e3d/pkgs/by-name/vo/voicevox-engine/pyopenjtalk.nix#L40-L44

and https://github.com/NixOS/nixpkgs/blob/a40440f13eb80881ab26734a36bb3005300e1e3d/pkgs/by-name/fa/fastcov/package.nix#L22-L27

Reply:
It depends on whether you need `python3Packages.cmake` or `pkgs.cmake`.

Reply:
Thank you. I didn't knew there is something Python specific in CMake.
It is used only for Ion C. So I guess it should be `pkgs.cmake`. I moved it.  
But I guess it still effectively pulls in `python3Packages.cmake` as in arguments it is just `cmake` alongside with `setuptools`.   
Does it mean I have to specify in arguments `pkgs` (or something similar) and write `pkgs.cmake`?

Reply:
You should put `inherit (pkgs) cmake` in `python-packages.nix`.